### PR TITLE
#167727626 Verify Email upon Signup

### PR DESCRIPTION
**Pull request by ChiamakaObitube**
#### What does this PR do?
This PR sends a verification email to a new user upon signup into Barefoot Nomad.

#### Description of Task to be completed?
* Write tests that will fail
* create a user model 
* create a verify user model
* create a helper method to use Sendgrid to sent email to user
* Test the user signup endpoint to ensure that a verification email is sent.

#### How should this be manually tested?
* Clone the repo and `cd` into the project directory
* Run `yarn install` to install the project's dependency
* Run `yarn start` to start the app
* Open Postman on `/auth/signup` 
* Pass in the following request parameters  ```firstName, lastName, email, password, isVerified``` as JSON objects
* S
* Enter a valid email address
* Upon signup, user will receiver an email to verify account.

#### What are the relevant pivotal tracker stories?
[##167727626](https://www.pivotaltracker.com/story/show/167727626)

#### Any background context you want to add?
Screenshot from 2019-08-22 22-08-39
WhatsApp Image 2019-08-22 at 22 12 51

### CODE REVIEW

File: src/utils/HelperMethods.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import sgMail from '@sendgrid/mail';
+// import Mailgen from 'mailgen';
+// import {config} from 'dotenv';
+// const sgMail = require('@sendgrid/mail');
+require('dotenv').config();
```
Comment (ChidozieGabriel) [inline src/utils/HelperMethods.js]: Line 5 is redundant. `dotenv.config()` is called once and from the entry file. That way, we wouldn't have to litter our code with many `dotenv.config()`s.

If you need more clarification check the dotenv [docs](https://www.npmjs.com/package/dotenv#usage) and maybe this [conversation](https://github.com/andela/rogue-backend/pull/23#discussion_r316191703)

Please ensure that `dotenv.config()` is called once and from the entry file (src/index.js).

File: src/migrations/20190822125944-create-verify-user.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('VerifyUsers', {
```
Comment (ChidozieGabriel) [inline src/migrations/20190822125944-create-verify-user.js]: We do not need another table to keep records of our verified users.

The boolean `isVerified` which you added in `src/migrations/20190814125500-create-user.js` lines 22-24 should suffice.

Add the properties of the `VerifyUsers` table to the `Users` table.
> A verified user is still a user

**Review by ChidozieGabriel — CHANGES_REQUESTED** [review]
Good work @ChiamakaObitube . Please attend to the following review.

File: src/utils/HelperMethods.js (hunk 1 above)
Comment (ChiamakaObitube) [reply src/utils/HelperMethods.js]: Hi Chidozie, thank you for your feedback.

**Pull request merged**